### PR TITLE
Add menu item model + ActiveAdmin UI

### LINKED ISSUE #1: Add ability to list and create new pages
The Admin::PagesController needs `#index`, `#new` and `#create` actions

### LINKED ISSUE #2: Add 'Menu' model
Based on the potential [design](https://trello-attachments.s3.amazonaws.com/58a4a2b8a51fe8a750712387/58f987f7c938ed15112e2e3c/8c9accfdef8c1f45947ea32a3600703d/v2_squared_buttons.png), the website should probably have a new model for storing Menu items. 

These could simply be a mapping of menu label to slug, with an `order` key to determine their presentation order. (Future fancy iterations will probably support drag and drop).

### PR DESCRIPTION
**Pull request by dominic**
Adds ActiveAdmin with the following models configured:

- Pages, with an optional "parent" relationship
- Menu Items, which can be dragged and dropped into a desired order

Fixes #1
Fixes #2

### CODE REVIEW

File: app/admin/admin.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+ActiveAdmin.register Admin do
+  permit_params :email, :password, :password_confirmation
```
Comment (john-olmsted-ck) [inline app/admin/admin.rb]: Is this a magical `activeadmin` DSL?

File: app/admin/dashboard.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    #       end
+    #     end
+    #   end
+
+    #   column do
+    #     panel "Info" do
+    #       para "Welcome to ActiveAdmin."
+    #     end
+    #   end
+    # end
```
Comment (john-olmsted-ck) [inline app/admin/dashboard.rb]: Do we want to just scrap these examples?

File: db/migrate/20171006050739_create_active_admin_comments.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class CreateActiveAdminComments < ActiveRecord::Migration::Current
+  def self.up
+    create_table :active_admin_comments do |t|
+      t.string :namespace
+      t.text   :body
+      t.references :resource, polymorphic: true
```
Comment (john-olmsted-ck) [inline db/migrate/20171006050739_create_active_admin_comments.rb]: What is `:resource` exactly? Some polymorphic abstraction that can have comments? A blog post would be a concrete example.

If my understanding is correct, could we consider naming this `:commentable`? I think that name would make it a little clearer, "resource" is maybe a little _too_ abstract.

File: app/admin/admin.rb (hunk 1 above)
Comment (qsymmachus) [inline app/admin/admin.rb]: Is this a magical `activeadmin` DSL?

File: db/migrate/20171006050739_create_active_admin_comments.rb (hunk 3 above)
Comment (qsymmachus) [inline db/migrate/20171006050739_create_active_admin_comments.rb]: What is `:resource` exactly? It's polymorphic, so something (anything!) that can have comments tied to it, right?

I would consider renaming this it `:commentable` to make this clearer – "resource" is a little _too_ abstract.

File: app/admin/admin.rb (hunk 1 above)
Comment (dominic) [reply app/admin/admin.rb]: Yeah, if `activeadmin` has one glaring flaw, it's the use of a custom DSL for everything.

This is it's implementation of `strong_params`: https://activeadmin.info/2-resource-customization.html#setting-up-strong-parameters

File: db/migrate/20171006050739_create_active_admin_comments.rb (hunk 3 above)
Comment (dominic) [reply db/migrate/20171006050739_create_active_admin_comments.rb]: This is `activeadmin` boilerplate, so I don't want to mess with it too much. It manifests as comments any logged in user can add to a model that shows up only in the admin interface:

screen shot 2017-10-06 at 10 51 44 pm

**Pull request merged**